### DISCUSSION TITLE
Issues - Permission denied

### DISCUSSION THREAD

**Discussion by cdtrinh** [adnanh/webhook#515] [Q&A] [answered]
Hi 

I just installed it and am trying to run a bash script.  The script works when I run it outside of the webhook. But when I initiate it with webhook, I get this. Everything is running in my home directory. 

trinsteradm1n@UbuntuVM:~$ /usr/bin/webhook -hooks block.json -verbose -port 9999[webhook] 2021/03/05 10:01:29 version 2.5.0 starting
[webhook] 2021/03/05 10:01:29 setting up os signal watcher
[webhook] 2021/03/05 10:01:29 attempting to load hooks from block.json
[webhook] 2021/03/05 10:01:29 found 1 hook(s) in file
[webhook] 2021/03/05 10:01:29   loaded: pihole-block
[webhook] 2021/03/05 10:01:29 serving hooks on http://0.0.0.0:9999/hooks/{id}
[webhook] 2021/03/05 10:01:29 os signal watcher ready
[webhook] 2021/03/05 10:01:31 Started GET /hooks/pihole-block
[webhook] 2021/03/05 10:01:31 pihole-block got matched
[webhook] 2021/03/05 10:01:31 pihole-block hook triggered successfully
[webhook] 2021/03/05 10:01:31 Completed 200 OK in 630.93µs
[webhook] 2021/03/05 10:01:31 executing /home/trinsteradm1n/block.sh (/home/trinsteradm1n/block.sh) with arguments ["/home/trinsteradm1n/block.sh"] and environment [] using /home/trinsteradm1n as cwd
[webhook] 2021/03/05 10:01:31 command output:
[webhook] 2021/03/05 10:01:31 error occurred: fork/exec /home/trinsteradm1n/block.sh: permission denied
[webhook] 2021/03/05 10:01:31 finished handling pihole-block

Here is my json file

[
  {
    "id": "pihole-block",
    "execute-command": "/home/trinsteradm1n/block.sh",
    "command-working-directory": "/home/trinsteradm1n"
  }
]

I also tried with sudo, but that did not work neither.

Here are the permissions in my home directory:
trinsteradm1n@UbuntuVM:~$ ls -la
total 76
drwxr-xr-x 6 trinsteradm1n trinsteradm1n  4096 Mar  5 10:04 .
drwxr-xr-x 3 root          root           4096 Feb 26 01:32 ..
-rw------- 1 trinsteradm1n trinsteradm1n  6823 Mar  5 09:36 .bash_history
-rw-r--r-- 1 trinsteradm1n trinsteradm1n   220 Apr  4  2018 .bash_logout
-rw-r--r-- 1 trinsteradm1n trinsteradm1n  3771 Apr  4  2018 .bashrc
-rw------- 1 trinsteradm1n trinsteradm1n 12288 Mar  5 09:36 .block.sh.swp
drwx------ 2 trinsteradm1n trinsteradm1n  4096 Feb 26 01:35 .cache
drwx------ 3 trinsteradm1n trinsteradm1n  4096 Feb 26 01:35 .gnupg
drwxrwxr-x 3 trinsteradm1n trinsteradm1n  4096 Mar  5 06:59 .local
-rw-r--r-- 1 trinsteradm1n trinsteradm1n   807 Apr  4  2018 .profile
drwx------ 2 trinsteradm1n trinsteradm1n  4096 Feb 26 01:32 .ssh
-rw-r--r-- 1 trinsteradm1n trinsteradm1n     0 Feb 26 01:36 .sudo_as_admin_successful
-rw------- 1 trinsteradm1n trinsteradm1n  8152 Mar  5 10:04 .viminfo
-rw-rw-r-- 1 trinsteradm1n trinsteradm1n   148 Mar  5 10:04 block.json
-rw-rw-r-- 1 trinsteradm1n trinsteradm1n     8 Mar  5 09:35 block.sh
-rw-rw-r-- 1 trinsteradm1n trinsteradm1n     5 Mar  5 09:40 test.txt

Please help!

**Comment by moorereason** [comment, marked as answer]
For the record, the answer is in https://github.com/adnanh/webhook/issues/516#issuecomment-791833356.
